File: zad2/PamoLab/app/src/main/java/com/example/pamolab/ui/BMI/BmiFragment.java
package com.example.pamolab.ui.BMI;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.ViewModelProvider;

import com.example.pamolab.databinding.FragmentBmiBinding;


public class BmiFragment extends Fragment {

    private BmiViewModel bmiViewModel;
    private FragmentBmiBinding binding;

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater,
                             ViewGroup container, Bundle savedInstanceState) {

        bmiViewModel = new ViewModelProvider(requireActivity()).get(BmiViewModel.class);

        binding = FragmentBmiBinding.inflate(inflater, container, false);
        View root = binding.getRoot();

        final TextView textView = binding.textViewResult;
        final EditText weightEditText = binding.editTextWeight;
        final EditText heightEditText = binding.editTextHeight;
        final Button calculateButton = binding.buttonCalculate;


        calculateButton.setOnClickListener(v -> {
            int weight = Integer.parseInt(weightEditText.getText().toString());
            int height = Integer.parseInt(heightEditText.getText().toString());

            // calculate BMI here
            double bmi = calculateBMI(weight, height);

            String result = String.format("Your BMI is %.1f", bmi);
            textView.setText(result);

            // store weight and height in view model
            bmiViewModel.setWeight(weight);
            bmiViewModel.setHeight(height);
        });

        return root;
    }

    private double calculateBMI(int weight, int height) {
        double heightMeters = height / 100.0;
        return weight / (heightMeters * heightMeters);
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        binding = null;
    }
}

File: zad4/PamoLab/app/src/main/java/com/example/pamolab/ui/BMI/BmiFragment.kt
package com.example.pamolab.ui.BMI

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.lifecycle.ViewModelProvider
import com.example.pamolab.databinding.FragmentBmiBinding

class BmiFragment : Fragment() {
    private var bmiViewModel: BmiViewModel? = null
    private var binding: FragmentBmiBinding? = null
    override fun onCreateView(inflater: LayoutInflater,
                              container: ViewGroup?, savedInstanceState: Bundle?): View? {
        bmiViewModel = ViewModelProvider(requireActivity()).get(BmiViewModel::class.java)
        binding = FragmentBmiBinding.inflate(inflater, container, false)
        val root: View = binding!!.root
        val textView = binding!!.textViewResult
        val weightEditText = binding!!.editTextWeight
        val heightEditText = binding!!.editTextHeight
        val calculateButton = binding!!.buttonCalculate
        calculateButton.setOnClickListener { v: View? ->
            val weight = weightEditText.text.toString().toInt()
            val height = heightEditText.text.toString().toInt()

            // calculate BMI here
            val bmi = calculateBMI(weight, height)
            val result = String.format("Your BMI is %.1f", bmi)
            textView.text = result

            // store weight and height in view model
            bmiViewModel!!.setWeight(weight)
            bmiViewModel!!.setHeight(height)
        }
        return root
    }

    private fun calculateBMI(weight: Int, height: Int): Double {
        val heightMeters = height / 100.0
        return weight / (heightMeters * heightMeters)
    }

    override fun onDestroyView() {
        super.onDestroyView()
        binding = null
    }
}